services/cms-price-check.js

```diff
@@ -0,0 +1,168 @@
+const express = require('express');
+const uuidv4 = require('uuid/v4');
+const priceTable = require('../lib/rxnorm-prices.json');
+
+const router = express.Router();
+
+function getValidCoding(codings, system) {
+  let validCoding = null;
+  codings.forEach((coding) => {
+    if (coding.code && coding.system === system) {
+      validCoding = coding;
+    }
+  });
+  return validCoding;
+}
+
+function getValidCodingFromConcept(medicationCodeableConcept) {
+  let coding = null;
+  if (medicationCodeableConcept && medicationCodeableConcept.coding) {
+    coding = getValidCoding(medicationCodeableConcept.coding, 'http://www.nlm.nih.gov/research/umls/rxnorm');
+  }
+  return coding;
+}
+
+function getValidResource(resource, context) {
+  let coding = null;
+  if (resource.resourceType === 'MedicationOrder' || resource.resourceType === 'MedicationRequest') {
+    // Check if patient in reference from medication resource refers to patient in context
+    if (resource.patient && resource.patient.reference === `Patient/${context.patientId}`) {
+      const { medicationCodeableConcept } = resource;
+      coding = getValidCodingFromConcept(medicationCodeableConcept);
+    }
+  }
+  return coding;
+}
+
+function getValidContextResources(request) {
+  const resources = [];
+  const { context } = request.body;
+  if (context && context.patientId && context.medications) {
+    const medResources = context.medications;
+    medResources.forEach((resource) => {
+      const validResource = getValidResource(resource, context);
+      if (validResource) {
+        resources.push(resource);
+      }
+    });
+  }
+  return resources;
+}
+
+function constructCard(summary, suggestionResource) {
+  const card = {
+    summary,
+    source: { label: 'CMS Public Use Files' },
+    indicator: 'info',
+  };
+
+  if (suggestionResource) {
+    card.suggestions = [{
+      label: 'Change to generic',
+      uuid: uuidv4(),
+      actions: [
+        {
+          type: 'create',
+          resource: suggestionResource,
+        },
+      ],
+    }];
+  }
+  return card;
+}
+
+function getSuggestionCard(prices, genericCode, resource) {
+  if (prices.generic && prices.brand) {
```

Comment:
Is `prices` guaranteed to be available ? 
If not checking for its existence would be nice to have : 
`if (prices && prices.generic && prices.brand) {`

Reply:
Yes, it's checked [here](https://github.com/cds-hooks/sandbox-cds-services/pull/7/files#diff-ad2cbff78178a04861ae72b427e24768R129) and we return null before the method is called

Reply:
👍